File: lab02/fun_rand.py
import random
import numpy as np

def exp(time_mean):
    a = 0.0
    while a == 0:
        a = random.random()
    a = -time_mean * np.log(a)
    return a


def uniform(time_min, time_max):
    a = 0.0
    while a == 0:
        a = random.random()
    a = time_min + a * (time_max - time_min)
    return a


def norm(time_mean, time_deviation):
    return time_mean + time_deviation * random.gauss(0.0, 1.0)

File: lab02/src/FunRand.java

import java.util.Random;

public  class FunRand {
    public static double exp(double timeMean) {
        double a = 0;
        while (a == 0) {
            a = Math.random();
        }
        a = -timeMean * Math.log(a);

        return a;
    }
    public static double unif(double timeMin, double timeMax) {
        double a = 0;
        while (a == 0) {
            a = Math.random();
        }
        a = timeMin + a * (timeMax - timeMin);

        return a;
    }
    public static double norm(double timeMean, double timeDeviation) {
        double a;
        Random r = new Random();
        a = timeMean + timeDeviation * r.nextGaussian();

        return a;
    }

}